rbd/locks.go

```diff
@@ -0,0 +1,120 @@
+//go:build !nautilus && ceph_preview
+// +build !nautilus,ceph_preview
+
+package rbd
+
+// #cgo LDFLAGS: -lrbd
+// #include <errno.h>
+// #include <stdlib.h>
+// #include <rbd/librbd.h>
+import "C"
+
+import (
+	"unsafe"
+)
+
+// LockMode represents a group of configurable lock modes.
+type LockMode C.rbd_lock_mode_t
+
+const (
+	// LockModeExclusive is the representation of RBD_LOCK_MODE_EXCLUSIVE from librbd.
+	LockModeExclusive = LockMode(C.RBD_LOCK_MODE_EXCLUSIVE)
+	// LockModeShared is the representation of RBD_LOCK_MODE_SHARED from librbd.
+	LockModeShared = LockMode(C.RBD_LOCK_MODE_SHARED)
+)
+
+// LockAcquire takes a lock on the given image as per the provided lock_mode
```

Comment:
nit: For better completeness you could even add a full-stop(`.`) towards the end of each API doc comment.